Fix test_wcs_nd_map_data_transpose_issue

I think here we forgot to pass `sparse=True` in `test_wcs_nd_map_data_transpose_issue`?

https://github.com/gammapy/gammapy/pull/1346/files#diff-930fa0f94b49b5d6537396fad7c6512aR137-R140

If I change to `sparse=True` in the write call, I get this fail:
```
E           Arrays are not equal
E           
E           x and y nan location mismatch:
E            x: array([[0., 1., 2.],
E                  [0., 0., 0.]], dtype=float32)
E            y: array([[  0.,   1.,   2.],
E                  [ nan,  inf, -inf]])
```

@adonath - Is there a bug?

Should we spend 1-2 days on maps basics and sort out dtype and missing value issues (#1440, #2374, maybe this issue), e.g. with pair-coding & debugging? Or defer all or some to v1.1?

This is a reminder issue, noticed this accidentally while working on
https://github.com/gammapy/gammapy/pull/2350#issuecomment-542754564

--- a/gammapy/maps/tests/test_wcsnd.py
+++ b/gammapy/maps/tests/test_wcsnd.py
@@ -149,7 +149,7 @@ def test_wcs_nd_map_data_transpose_issue(tmp_path):
     assert_equal(m2.data, data)
 
     # Data should be unmodified after write / read to sparse image format
-    m.write(tmp_path / "sparse.fits.gz")
+    m.write(tmp_path / "sparse.fits.gz", sparse=True)
     m2 = Map.read(tmp_path / "sparse.fits.gz")
     assert_equal(m2.data, data)
 

--- a/gammapy/maps/wcsmap.py
+++ b/gammapy/maps/wcsmap.py
@@ -270,18 +270,16 @@ class WcsMap(Map):
 
         if len(shape) == 2:
             data_flat = np.ravel(data)
-            data_flat[~np.isfinite(data_flat)] = 0
-            nonzero = np.where(data_flat > 0)
-            value = data_flat[nonzero].astype(float)
+            non_zero = np.where(~(data_flat == 0))
+            value = data_flat[non_zero].astype(float)
             cols = [
-                fits.Column("PIX", "J", array=nonzero[0]),
+                fits.Column("PIX", "J", array=non_zero[0]),
                 fits.Column("VALUE", "E", array=value),
             ]
         elif npix[0].size == 1:
             shape_flat = shape[:-2] + (shape[-1] * shape[-2],)
             data_flat = np.ravel(data).reshape(shape_flat)
-            data_flat[~np.isfinite(data_flat)] = 0
-            nonzero = np.where(data_flat > 0)
+            nonzero = np.where(~(data_flat == 0))
             channel = np.ravel_multi_index(nonzero[:-1], shape[:-2])
             value = data_flat[nonzero].astype(float)
             cols = [
@@ -295,8 +293,7 @@ class WcsMap(Map):
             pix = []
             for i, _ in np.ndenumerate(npix[0]):
                 data_i = np.ravel(data[i[::-1]])
-                data_i[~np.isfinite(data_i)] = 0
-                pix_i = np.where(data_i > 0)
+                pix_i = np.where(~(data_i == 0))
                 data_i = data_i[pix_i]
                 data_flat += [data_i]
                 pix += pix_i

